### PR TITLE
Feature/prog2 22 refactor

### PR DESCRIPTION
**Pull request by sorinlzr**
- add watchlist view
- implement logic for adding and removing movies from watchlist
- add error handling
- refactor several files

### CODE REVIEW

**Review by burakkongo — DISMISSED** [review]
Lgtm. Good job man. Pula

File: src/main/java/at/ac/fhcampuswien/fhmdb/controller/WatchlistController.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    @FXML
+    private StackPane parent;
+    @FXML
+    protected JFXButton homeButton;
+
+
+    @Override
+    public void initialize() {
+        super.initialize();
```
Comment (benniii1) [inline src/main/java/at/ac/fhcampuswien/fhmdb/controller/WatchlistController.java:26]: ### Remove Button not shown in the Movie-Cell

Add this before super.initialize():
`this.isWatchlistCell = true;`

File: src/main/java/at/ac/fhcampuswien/fhmdb/controller/AbstractViewController.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    protected SVGPath burger = new SVGPath();
+    protected final ObservableList<Movie> movies = FXCollections.observableArrayList();
+
+    protected ClickEventHandler<Movie> onWatchlistButtonClicked;
+
+    protected WatchlistRepository repository;
+
+    public static final String NO_DB_CONNECTION_AVAILABLE = "No database connection available";
+
```
Comment (benniii1) [inline src/main/java/at/ac/fhcampuswien/fhmdb/controller/AbstractViewController.java:54]: ### Remove Button not shown in the Movie-Cell

Add the following code here:
`protected boolean isWatchlistCell = false;`

File: src/main/java/at/ac/fhcampuswien/fhmdb/controller/AbstractViewController.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        burger.setStrokeWidth(4);
+
+        cross.setContent(SVG.CROSS);
+        cross.setStroke(Color.WHITE);
+        cross.setStrokeWidth(4);
+
+        navigationButton.setGraphic(burger);
+
+        movieListView.setItems(movies);
+        movieListView.setCellFactory(e -> new MovieCell(onWatchlistButtonClicked));
```
Comment (benniii1) [inline src/main/java/at/ac/fhcampuswien/fhmdb/controller/AbstractViewController.java:79]: ### Remove Button not shown in the Movie-Cell

Refactor to this
`movieListView.setCellFactory(e -> new MovieCell(onWatchlistButtonClicked, isWatchlistCell));`

File: src/main/java/at/ac/fhcampuswien/fhmdb/ui/MovieCell.java [hunk 4]
```diff
@@ -9,35 +10,45 @@
 
 public class MovieCell extends ListCell<Movie> {
 
+    private final ClickEventHandler<Movie> clickEventHandler;
+
```
Comment (benniii1) [inline src/main/java/at/ac/fhcampuswien/fhmdb/ui/MovieCell.java:14]: ### Remove Button not shown in the Movie-Cell

Add this here:
`private final boolean isWatchlistCell;`

File: src/main/java/at/ac/fhcampuswien/fhmdb/ui/MovieCell.java [hunk 5]
```diff
@@ -9,35 +10,45 @@
...
+                controller.setTitle(movie.getTitle());
+                controller.setDescription(movie.getDescription());
+                controller.setGenre(movie.getGenres());
+                controller.setYear(movie.getReleaseYear());
+                controller.setRuntime(movie.getLengthInMinutes());
+                controller.setRating(movie.getRating());
+                controller.setWriters(movie.getWriters());
+                controller.setDirectors(movie.getDirectors());
+                controller.setMainCast(movie.getMainCast());
+                setGraphic(graphic);
```
Comment (benniii1) [inline src/main/java/at/ac/fhcampuswien/fhmdb/ui/MovieCell.java:47]: ### Remove Button not shown in the Movie-Cell

Add this before setGraphic(graphic)
`if (isWatchlistCell) controller.addToWatchlist();`

File: src/main/java/at/ac/fhcampuswien/fhmdb/ui/MovieCell.java [hunk 6]
```diff
@@ -9,35 +10,45 @@
 
 public class MovieCell extends ListCell<Movie> {
 
+    private final ClickEventHandler<Movie> clickEventHandler;
+
+    public MovieCell(ClickEventHandler<Movie> clickEventHandler) {
+        super();
+        this.clickEventHandler = clickEventHandler;
+    }
+
```
Comment (benniii1) [inline src/main/java/at/ac/fhcampuswien/fhmdb/ui/MovieCell.java:19]: ### Remove Button not shown in the Movie-Cell

Change to:
`public MovieCell(ClickEventHandler<Movie> clickEventHandler, boolean isWatchlistCell) {
        super();
        this.clickEventHandler = clickEventHandler;
        this.isWatchlistCell = isWatchlistCell;
    }`

**Review by benniii1 — CHANGES_REQUESTED** [review]
Looks good! Cool that you extracted the separate views to an abstract class and reused it and found out how to make the navigation to be an overlay :)

However, the movie-cell does not show the Remove field in the watchlist-view. I added the needed code in the comments of the review.

**Review by benniii1 — APPROVED** [review]

**Pull request merged**